torch/autograd/gradcheck.py

```diff
@@ -225,12 +274,24 @@ def fn_pack_inps(*inps):
 def _compute_numerical_gradient(fn, entry, v, norm_v, nbhd_checks_fn):
     # Performs finite differencing by perturbing `entry` in-place by `v` and
     # returns the gradient of each of the outputs wrt to x at idx.
-    orig = entry.clone()
-    entry.copy_(orig - v)
-    outa = fn()
-    entry.copy_(orig + v)
-    outb = fn()
-    entry.copy_(orig)
+    if _is_sparse_compressed_tensor(entry):
+        # sparse compressed tensors don't implement sub/add/copy_ yet.
+        assert entry.layout == v.layout, (entry.layout, v.layout)
+        assert entry._nnz() == v._nnz(), (entry._nnz(), v._nnz(), entry.shape)
+        entry = entry.detach()
```

Comment:
Out of curiosity, why do we need to detach here?

If it is something temporary (as the comment suggests), maybe keeping it here is probably fine, but normally this logic might belong in _iter_tensor (for the slow gradcheck path) and _get_input_to_perturb (for the fast gradcheck path).